bolt-design-system/bolt
# Video Cue Points

## Jira
http://vjira2:8080/browse/BDS-681?filter=-1 and http://vjira2:8080/browse/BDS-673

## Summary
Enhances the `<bolt-video>` component to support displaying embedded calls to action (Cue Points) while a video is supported by a particular player and video playing. Also adds support for the new plugin system to allow plugins to get automatically added / removed from a particular video player instance automatically via the `<bolt-video>` custom element.

## Details
- Refactors how video player handles loading and enabling plugins so the code is now compiled, checked for errors, and validated like any other code being shipped in the design system
- Adds the ability to manually opt in and opt out of any of the available video plugins based on the new `enabled-plugins` and `disabled-plugins` props
- Added internal checks to ensure video player plugins aren't accidentally double-loaded
- Also adds continued support for having some video player plugins load up via the original `on-init` inline script approach in addition to using the new built-in plugins being made available.
- Added the new `cue` video player plugin + internal logic to handle adding, removing, and rendering embedded video player CTAs; includes support for multiple video player instances in addition to support for much more consistently displaying these if the video player is paused or scrubbed.
- Refactors / moves the existing video player meta sub-component to live as a standalone sub-component used internally in the video player (separate files = less logic crammed into a single file)
- Also cleans up existing util functions used with the video player and bumps them out into their own standalone utils helper used by the video component

## How to test
1. Confirm all examples displayed on the view-all video player page work as expected
2. Confirm multiple video player instances on the same page can display the new cue points functionality (each player being self-contained)
3. Confirm the original video player plugin approach continues to work as expected (ie. no mandatory backend changes would be required with these updates)
4. Verify the recent translated share text updates from @remydenton continue to work as expected
